dask_ml/model_selection/_incremental.py

```diff
@@ -347,3 +354,228 @@ def fit(
+            scoring=scoring,
+            iid=iid,
+            refit=refit,
+            cv=cv,
+            error_score=error_score,
+            return_train_score=return_train_score,
+            scheduler=scheduler,
+            n_jobs=n_jobs,
+            cache_cv=cache_cv,
+        )
+
+    def _get_cv(self, X, y):
+        X_train, X_test, y_train, y_test = train_test_split(
+            X, y, test_size=self.test_size
+        )
+        return X_train, X_test, y_train, y_test
+
+    def _get_ids_and_args(self):
+        # type: () -> Dict[int, Tuple(List[Dict], Callable)]
+        """Abstract method for generating the argumnets passed to `fit`.
+
+        Returns
+        -------
+        ids_and_args : Dict
+            Should have integers for keys and the following arguments
+
+            * param_list : List[Dict]
+            * additional_fit_calls : Callable
+        """
+        raise NotImplementedError
+
+    def _get_cv_results(self, results):
+        # type: (Dict) -> Dict
+        """Construct the CV results.
+
+        Should have the following keys:
+
+        * params
+        * test_score
+        * mean_test_score
+        * rank_test_score
+        * mean_partial_fit_time
+        * std_partial_fit_time
+        * mean_score_time
+        * std_score_time
+        * partial_fit_calls
+        * model_id
+        """
+        raise NotImplementedError
+
+    def _get_best_estimator(self, results):
+        # type: (Dict) -> Estimator
+        """Select the best estimator from the set of estimators."""
+        raise NotImplementedError
+
+    def _update_results(self, results):
+        # type: (Dict) -> None
+        """Update `self` with attributes extracted from `results`."""
+        raise NotImplementedError
+
+    def fit(self, X, y, **fit_params):
+        """Find the best parameters for a particular model
+        Parameters
+        ----------
+        X, y : array-like
+        **fit_params
+            Additional partial fit keyword arguments for the estimator.
+        """
+        ids_and_args = self._get_ids_and_args()
+
+        X_train, X_test, y_train, y_test = self._get_cv(X, y)
+        self.scorer_ = check_scoring(self.estimator, scoring=self.scoring)
+
+        results = {
+            model_id: fit(
+                self.estimator,
+                param_list,
+                X_train,
+                y_train,
+                X_test,
+                y_test,
+                additional_calls=additional_calls,
+                fit_params=fit_params,
+                scorer=self.scorer_,
+                random_state=self.random_state,
+            )
+            for model_id, (param_list, additional_calls) in ids_and_args.items()
+        }
+        self._update_results(results)
+        self.cv_results_ = self._get_cv_results(results)
+        self.best_estimator_ = self._get_best_estimator(results)
+        return self
+
+
+class IncrementalRandomizedSearchCV(BaseIncrementalSearchCV):
```

Comment:
There are two parts to this class: the method of choosing parameters, and the incremental/adaptive method implemented (currently `remove_worst`). I'd rather the name reflect the incremental/adaptive component: maybe something like `KillWorstModelCV`?

[1]:https://pytorch.org/docs/stable/optim.html#torch.optim.lr_scheduler.ReduceLROnPlateau

Reply:
I think that this is only up here as a strawman to see what things look like when extending the class.  I don't think that the `remove_worst` function is a particularly pragmatic approach.  It was just something simple that was small enough for me to put into a docstring.